src/service/resources-impl.js

```diff
@@ -1402,6 +1408,37 @@ export class Resources {
       // Selection API not supported.
     }
   }
+
+  /**
+   * Cleanup task queues from tasks for elements that has been unloaded.
+   * @param resource
+   * @private
+   */
+  maybeRemoveFromQueues_(resource) {
+    if (resource.getState() == ResourceState_.NOT_LAID_OUT) {
+      // If the layout promise for this resource has not resolved yet, remove
+      // it from the task queues to make sure this resource can be rescheduled
+      // for layout again later on.
+      // TODO(mkhatib): Look into if we need to do the same for preload tasks, this
+      // depends on whether we expect people to unload prerendered DOM.
+      // The answer might be no since we probably want to keep the prerendered
+      // DOM around to use as placeholder when the user navigate back to the
+      // document.
+      const taskId = resource.getTaskId(LAYOUT_TASK_ID_);
```

Comment:
As number of types of tasks grows - this will be less and less tenable. I think we should just iterate through queues and remove found entries via `resource == task.resource`. The queues should be small - that's the point. If we have concerns about the performance then the first best place would be to always add API to TaskQueue: `.dequeueByElement()` or such where we can freely optimize the implementation and unit-test separately.